### PR TITLE
*: wrap the previous statement for performance

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
Formating a long previous statement takes too many resources, which may hurt the performance.

### What is changed and how it works?
Warp it to a `fmt.Stringer`.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: util/stringutil/string_util.go [hunk 1]
```diff
@@ -252,17 +252,17 @@ func Copy(src string) string {
 	return string(hack.Slice(src))
 }
 
-// stringerFunc defines string func implement fmt.Stringer.
-type stringerFunc func() string
+// StringerFunc defines string func implement fmt.Stringer.
```
Comment (lonng) [inline util/stringutil/string_util.go]: I think it's better to define a cachable struct `struct{cacheStr string; fn func() string}`, the performance will be improved if the `StringFunc` will be called many times.

Comment (jackysp) [reply util/stringutil/string_util.go]: This is generally not used multiple times. Maybe we can improve it later.

**Comment by jackysp** [comment]
PTAL @lonng

**Comment by jackysp** [comment]
Wide table batch insert case:
```
Version:
tidb: ed610881ac14e235d0ab3c0092276dfb226a0a29(HEAD) 2019-09-19 04:04:18
tikv: 15082b6cb7d1be3414badc0b74a21b67a505a29e(master) 2019-09-20 05:47:23
pd: 4ca2265d180531191dcfc531e468ab8380737a33(master) 2019-09-18 08:29:29

Compare:
tidb: https://github.com/pingcap/tidb/compare/8cb488a...ed61088
tikv: https://github.com/tikv/tikv/compare/15082b6...15082b6
pd: https://github.com/pingcap/pd/compare/4ca2265...4ca2265

tidb_max_cpu: 22.76, delta: -0.62%
tikv_max_cpu: 1.77, delta: 1.84%
tidb_max_memory: 543.75 MiB, delta: -1.04%
tikv_max_memory: 1220.92 MiB, delta: -8.27%

tps: 4304.40 ± 0.5379% (std=12.72) delta: 2.76%
qps: 17577.92 ± 0.4999% (std=44.58) delta: 2.76%
lat_avg: 59.47 ± 0.5263% (std=0.18) delta: -2.69%
lat_99th: 170.48 ± 0.0000% (std=0.00) delta: 1.82%
```

**Pull request merged**